### ISSUE TITLE
mkfs.xfs fails to create grootfs on local-ssd (nvme) 

### ISSUE THREAD

**Issue by hprotzek** [cloudfoundry/garden-runc-release#129]
We are running concourse workers on gcp with ephemeral local-ssd. (PR for the Google CPI https://github.com/cloudfoundry/bosh-google-cpi-release/pull/305) Unfortunately the included version of xfsprogs is unable to create the grootfs on the ephemeral local-ssd:

```
{"timestamp":"1551798070.482636452","source":"grootfs","message":"grootfs.init-store.store-manager-init-store.existing-backing-store-could-not-be-mounted: Mounting filesystem: exit status 32: mount: wrong fs type, bad option, bad superblock on /dev/loop1,\n       missing codepage or helper program, or other error\n\n       In some cases useful info is found in syslog - try\n       dmesg | tail or so.\n","log_level":1,"data":{"session":"1.1","spec":{"UIDMappings":[{"HostID":4294967294,"NamespaceID":0,"Size":1},{"HostID":1,"NamespaceID":1,"Size":4294967293}],"GIDMappings":[{"HostID":4294967294,"NamespaceID":0,"Size":1},{"HostID":1,"NamespaceID":1,"Size":4294967293}],"StoreSizeBytes":388486316032},"storePath":"/var/vcap/data/grootfs/store/unprivileged"}}
{"timestamp":"1551798070.486574173","source":"grootfs","message":"grootfs.init-store.store-manager-init-store.overlayxfs-init-filesystem.formatting-filesystem.formatting-filesystem-failed","log_level":2,"data":{"cmd":["mkfs.xfs","-f","/var/vcap/data/grootfs/store/unprivileged.backing-store"],"error":"exit status 1","filesystemPath":"/var/vcap/data/grootfs/store/unprivileged.backing-store","session":"1.1.2.1","spec":{"UIDMappings":[{"HostID":4294967294,"NamespaceID":0,"Size":1},{"HostID":1,"NamespaceID":1,"Size":4294967293}],"GIDMappings":[{"HostID":4294967294,"NamespaceID":0,"Size":1},{"HostID":1,"NamespaceID":1,"Size":4294967293}],"StoreSizeBytes":388486316032},"stderr":"existing superblock read failed: Invalid argument\nmkfs.xfs: pwrite64 failed: Unknown error -22\n","stdout":"meta-data=/var/vcap/data/grootfs/store/unprivileged.backing-store isize=512    agcount=4, agsize=23711323 blks\n         =                       sectsz=512   attr=2, projid32bit=1\n         =                       crc=1        finobt=1, sparse=0\ndata     =                       bsize=4096   blocks=94845292, imaxpct=25\n         =                       sunit=0      swidth=0 blks\nnaming   =version 2              bsize=4096   ascii-ci=0 ftype=1\nlog      =internal log           bsize=4096   blocks=46311, version=2\n         =                       sectsz=512   sunit=0 blks, lazy-count=1\nrealtime =none                   extsz=4096   blocks=0, rtextents=0\n","storePath":"/var/vcap/data/grootfs/store/unprivileged"}}
{"timestamp":"1551798070.486697674","source":"grootfs","message":"grootfs.init-store.store-manager-init-store.initializing-filesystem-failed","log_level":2,"data":{"backingstoreFile":"/var/vcap/data/grootfs/store/unprivileged.backing-store","error":"Formatting XFS filesystem: exit status 1","session":"1.1","spec":{"UIDMappings":[{"HostID":4294967294,"NamespaceID":0,"Size":1},{"HostID":1,"NamespaceID":1,"Size":4294967293}],"GIDMappings":[{"HostID":4294967294,"NamespaceID":0,"Size":1},{"HostID":1,"NamespaceID":1,"Size":4294967293}],"StoreSizeBytes":388486316032},"storePath":"/var/vcap/data/grootfs/store/unprivileged"}}
{"timestamp":"1551798070.486739635","source":"grootfs","message":"grootfs.init-store.init-store-failed","log_level":2,"data":{"error":"initializing filesyztem: Formatting XFS filesystem: exit status 1","session":"1"}}
```

Updating xfsprogs to the latest version 4.20.0 fixes the problem and garden is able to create the grootfs.

**Comment by BooleanCat** [comment]
@Callisto13 we had an issue with xfs progs in the past I believe - was the issue that we relied on the version shipped with the stemcell rather than bundling the version that we do?

I can't think of a reason not to just bump to latest.

**Comment by Callisto13** [comment]
Aaah yes that mess! 😂 
Should be fine to bump (pending deploy, tests etc etc)

**Comment by Callisto13** [comment]
Fixed in 1.19.1.
Closing.
